examples/md-flexible/src/Simulation.h

```diff
@@ -650,3 +679,62 @@ double Simulation<Particle>::calculateHomogeneity(autopas::AutoPas<Particle> &au
   // finally calculate standard deviation
   return sqrt(variance);
 }
+
+template <class Particle>
+void Simulation<Particle>::printProgress(size_t iterationProgress, size_t maxIterations, bool maxIsPrecise) {
+  // percentage of iterations complete
+  double fractionDone = static_cast<double>(iterationProgress) / maxIterations;
+
+  // length of the number of maxIterations
+  size_t numCharsOfMaxIterations = std::to_string(maxIterations).size();
+
+  // trailing information string
+  std::stringstream info;
+  info << std::setw(3) << std::round(fractionDone * 100) << "% " << std::setw(numCharsOfMaxIterations)
+       << iterationProgress << "/";
+  if (not maxIsPrecise) {
+    info << "~";
+  }
+  info << maxIterations;
+
+  // actual progress bar
+  std::stringstream progressbar;
+  progressbar << "[";
+  // get current terminal width
+  struct winsize w;
+  ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
+  auto terminalWidth = w.ws_col;
+  // the bar should fill the terminal window so subtract everything else (-2 for "] ")
+  int maxBarWidth = terminalWidth - info.str().size() - progressbar.str().size() - 2;
+  // sanity check
+  if (maxBarWidth < 1) {
+    std::cerr << "Warning! Terminal width appears to be too small or could not be read. Disabling progress bar."
+              << std::endl;
+    _config->dontShowProgressBar.value = true;
+    return;
+  }
+  auto barWidth = std::max(std::min(static_cast<decltype(maxBarWidth)>(maxBarWidth * (fractionDone)), maxBarWidth), 1);
+  // don't print arrow tip if >= 100%
+  if (iterationProgress >= maxIterations) {
+    progressbar << std::string(barWidth, '=');
+  } else {
+    progressbar << std::string(barWidth - 1, '=') << '>' << std::string(maxBarWidth - barWidth, ' ');
+  }
+  progressbar << "] ";
+  // clear current line (=delete previous progress bar)
+  std::cout << std::string(terminalWidth, '\r');
+  // print everything
+  std::cout << progressbar.str() << info.str() << std::flush;
+}
+template <class Particle>
+std::tuple<size_t, bool> Simulation<Particle>::estimateNumIterations() const {
+  if (_config->tuningPhases.value > 0) {
+    // @TODO: this can be improved by considering the tuning strategy
+    // 30 is just a random number for the number of tested configurations per tuning phase
+    auto estimate = (_config->tuningPhases.value - 1) * _config->tuningInterval.value +
+                    (_config->tuningPhases.value * _config->tuningSamples.value * 30);
```

Comment:
```suggestion
    // estimatedNumConfigs is just a random number for the number of tested configurations per tuning phase.
    size_t estimatedNumConfigs = 30;
    auto estimate = (_config->tuningPhases.value - 1) * _config->tuningInterval.value +
                    (_config->tuningPhases.value * _config->tuningSamples.value * estimatedNumConfigs);
```

30 appears to be a bit small for fullsearch (the default):
![image](https://user-images.githubusercontent.com/38119442/108503521-ce8da080-72b4-11eb-9a62-f5df9bdf4833.png)
and a bit big for bayesian-search:
![image](https://user-images.githubusercontent.com/38119442/108503755-317f3780-72b5-11eb-964a-ed8bcce33b76.png)

but, well, doesn't matter anyways...